packages/edit-site/src/components/dataviews/view-table.js

```diff
@@ -257,6 +262,69 @@ function ViewTable( {
 			}
 			return column;
 		} );
+		if ( selection !== undefined ) {
+			_columns.unshift( {
+				header: (
+					<CheckboxControl
+						__nextHasNoMarginBottom
+						checked={ areAllSelected }
+						indeterminate={ ! areAllSelected && selection.length }
+						onChange={ () => {
+							if ( areAllSelected ) {
+								setSelection( [] );
+							} else {
+								setSelection( data.map( ( { id } ) => id ) );
+							}
+						} }
+						label={
+							areAllSelected
+								? __( 'Deselect all' )
+								: __( 'Select all' )
+						}
+					/>
+				),
+				id: 'selection',
+				cell: ( props ) => {
+					//console.log({ props });
+					const item = props.row.original;
+					const isSelected = selection.includes( item.id );
+					//console.log({ item, isSelected });
+					let selectionLabel;
+					if ( getSelectionLabel ) {
+						selectionLabel = getSelectionLabel( isSelected, item );
+					} else {
+						selectionLabel = isSelected
+							? __( 'Deselect item' )
+							: __( 'Select a new item' );
+					}
+					return (
+						<CheckboxControl
+							__nextHasNoMarginBottom
+							checked={ isSelected }
+							label={ selectionLabel }
+							onChange={ () => {
+								if ( ! isSelected ) {
+									const newSelection = [
+										...selection,
+										item.id,
+									];
+									setSelection( newSelection );
+								} else {
+									setSelection(
+										selection.filter(
+											( id ) => id !== item.id
+										)
+									);
+								}
+							} }
+						/>
+					);
+				},
+				enableHiding: false,
+				width: 40,
+				className: 'dataviews-table-view__selection-column',
```

Comment:
What do you think of scoping the CSS using the `data-field-id` that we already [inject](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/dataviews/view-table.js#L475)? We target the [actions column](https://github.com/WordPress/gutenberg/blob/43c3d854d5f42b2700065873197a1bdade146e8c/packages/edit-site/src/components/dataviews/style.scss#L44) using this mechanism. It'd be good to use the same mechanism for all.

Also, I don't like that we expand the Field API to allow `className` for fields (the change below to support this class opens it up for every field).